Added support of Nucleo F401RE

LaurentLM opened the pull request:
Option for Nucleo F401RE board added. It seems to work the same as F446RE on my machine. 

I copied F411 oscillator settings in Src/main.c (line 139 onwards). It may be worth a second look if you know about these things.

terjeio commented on Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:
```diff
@@ -73,7 +73,7 @@
                                    STM32F439NI, STM32F439IG and STM32F439II Devices */
   /* #define STM32F401xC */   /*!< STM32F401CB, STM32F401CC, STM32F401RB, STM32F401RC, STM32F401VB and STM32F401VC Devices */
-  /* #define STM32F401xE */   /*!< STM32F401CD, STM32F401RD, STM32F401VD, STM32F401CE, STM32F401RE and STM32F401VE Devices */
+   #define STM32F401xE    /*!< STM32F401CD, STM32F401RD, STM32F401VD, STM32F401CE, STM32F401RE and STM32F401VE Devices */
```
This file should not be modified, instead add a build configurations (debug and release) and set the symbol there. Right click on the project and select _Build configurations > Manage..._ then _New..._ Copy the settings from F401 Blackpill and change the mcu symbol afterwards.

terjeio reviewed (changes requested):
Please add build configurations instead of modifying stm32f4xx.h as this messes up building for all the other variants.

LaurentLM commented:
Hi Terje,
Thanks for the feedback. I've reverted the file to original. 
I already had a debug configuration with the right symbols, I've added a release one.

LaurentLM commented:
> 
> 
> Hi Terje, Thanks for the feedback. I've reverted the file to original. I already had a debug configuration with the right symbols, I've added a release one.

Note: I've also removed the launch file as this doesn't seem to be needed. I'm pretty sure it wasnt clean anyway.

terjeio commented:
I've moved the clock init down to the last #else for STM32F401 and added `__HAL_PWR_VOLTAGESCALING_CONFIG` which was missing. I have also added a new symbol `IS_NUCLEO_DEVKIT` to simplify processing of guarded code.
